### PR TITLE
17 redo database

### PR DESCRIPTION
**Pull request by Estevenson1994**
create posts and likes databases/migrations and added profilepic/firstname etc to Users

### CODE REVIEW

**Review by david-mears — APPROVED** [review]

File: db/migrate/20190227122556_add_user_to_post.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserToPost < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :posts, :user, foreign_key: true
```
Comment (ch359) [inline db/migrate/20190227122556_add_user_to_post.rb]: Think we need to add something like `null:false` to the migration files

File: db/migrate/20190227123415_add_user_to_like.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserToLike < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :likes, :user, foreign_key: true
```
Comment (ch359) [inline db/migrate/20190227123415_add_user_to_like.rb]: Also potentially needs `null:false` to ensure users are linked to likes

File: db/migrate/20190227123437_add_post_to_like.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddPostToLike < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :likes, :post, foreign_key: true
```
Comment (ch359) [inline db/migrate/20190227123437_add_post_to_like.rb]: Ditto for linking to posts

**Review by ch359 — CHANGES_REQUESTED** [review]
Please could we add some checking to disallow null values for user ID and post ID?
The foreign key stops creating (e.g.) posts by a user that doesn't exist, but it is possible to make orphan posts associated with no user (i.e. null is allowed). The same applies to the two constraints in the Likes table.

E.g. check out the right-most column:
image

**Review by ch359 — APPROVED** [review]

**Review by david-mears — APPROVED** [review]

**Pull request merged**